### ISSUE TITLE
Move budget and plan properties in data processing requests to extension

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-api#544]
Move the budget and plan properties in data processing requests to extension.
We should make it identifiable whether these properties are supported or not as they are not very commonly supported.
Makes the interface cleaner and more predictable.

This will allow to provide an additional conformance class and makes it easier for clients to identify whether this is supported.
Alternatively, we could also expose the two parameters as "job options", see #471

**Comment by soxofaan** [comment]
Good idea. I think it makes sense to decouple the cost/budget aspects from the core openEO API

**Comment by m-mohr** [comment]
Will create a PR, but I'm waiting for the other PRs to be merged to avoid merge conflicts.
